dev/com.ibm.ws.cdi.internal/src/com/ibm/ws/cdi/internal/archive/liberty/ExtensionArchiveImpl.java

```diff
@@ -70,6 +78,7 @@ public boolean isExtClassesOnly() {
     public Set<String> getExtensionClasses() {
         Set<String> extensionClasses = super.getExtensionClasses();
         extensionClasses.addAll(spiExtensions);
+        extensionClasses.addAll(spiBuildCompatibleExtensions);
         return extensionClasses;
     }
```

Comment:
What is `getExtensionClasses()` used for?

Is it correct to have build compatible extension classes in here?

Reply:
`getExtensionClasses` is used by BDAFactory to pass in the extensionClassNames to BeanDeploymentArchiveImpl's constructor. After that there's code in scanForBeanClassNames that can include it in `beanClasses`. Which gets passed up to weld via `getBeanClasses`

I've read both the javadoc and the integration guide for what weld expects `getBeanClasses` and it was pretty terrible, but it looks like it expects results based on the bean scanning mode which is defined in the CDI spec, and checking the 4.0 spec I can't see any reason to treat BCE and regular Es differently.

Reply:
As discussed, we don't think the spiBuildCompatibleExtensions should be added here.

However, we might need the `LiteExtensionTranslator` class added. I think some parts of BeanDeploymentArchiveImpl assume we can tell if an archive is an extension archive by seeing if it has either an Extension class or a BCE class.